docroot/modules/custom/mass_content/src/Commands/MassContentCommands.php

```diff
@@ -126,4 +126,95 @@ public function migrateDateFields(string $type = '', int $limit = 0) {
 
   }
 
+  /**
+   * Migrate Service data.
+   *
+   * @param int $limit
+   *   Number of nodes to process
+   *   Argument provided to the drush command.
+   *
+   * @command mass-content:migrate-service
+   *
+   * @usage mass-content:migrate-service 1000
+   *   1000 is the number of nodes that will be processed.
+   */
+  public function migrateServiceData(int $limit = 0) {
+    $fields = [
+      'field_service_ref_actions_2',
+      'field_link_group',
+      'field_service_ref_actions',
+      'field_service_key_info_links_6',
+      'field_template',
+      'field_service_ref_locations'
+    ];
+    // 1. Log the start of the script.
+    $this->loggerChannelFactory->get('mass_content')->info('Update nodes batch operations start');
+
+    // 2. Retrieve all nodes of this type.
+    $storage = $this->entityTypeManager->getStorage('node');
+    try {
+      $query = $storage->getQuery();
+      $query->condition('type', 'service_page');
+      $orCondition = $query->orConditionGroup();
+      foreach ($fields as $field) {
+        $and = $query->andConditionGroup();
+        $and->exists($field);
+        $orCondition->condition($and);
+      }
+      $query->condition($orCondition);
+      $query->sort('nid');
+      if ($limit !== 0) {
+        $query->range(0, $limit);
+      }
+
+      $nids = $query->execute();
+    }
+    catch (\Exception $e) {
+      $this->output()->writeln($e);
+      $this->loggerChannelFactory->get('mass_content')->error('Error found @e', ['@e' => $e]);
```

Comment:
Are you intending to log an eror and keep going? Thats what this does. ALso, I think you want e->getMessage() at the end. Otherwise you are not passing a string but rather an object of type Exception.

Reply:
Fixed.